File: app/src/main/java/com/example/milestonesapplication/views/MilestoneMarker.java
package com.example.milestonesapplication.views;

import android.text.TextUtils;

import com.example.milestonesapplication.model.Milestone;
import com.google.android.gms.maps.model.LatLng;
import com.google.maps.android.clustering.ClusterItem;

public class MilestoneMarker implements ClusterItem {

    private final Milestone milestone;

    public MilestoneMarker(Milestone milestone) {
        this.milestone = milestone;
    }

    @Override
    public LatLng getPosition() {
        return milestone.getLocation();
    }

    @Override
    public String getTitle() {
        return milestone.getDescription();
    }

    @Override
    public String getSnippet() {
        return null;
    }

    public String getDescription() {
        return milestone.getName();

    }
}

File: app/src/main/java/com/example/milestonesapplication/views/MilestoneMarker.kt
package com.example.milestonesapplication.views

import com.example.milestonesapplication.models.Milestone
import com.google.android.gms.maps.model.LatLng
import com.google.maps.android.clustering.ClusterItem

class MilestoneMarker(private val milestone: Milestone) : ClusterItem {

    val description: String?
        get() = milestone.name

    override fun getPosition(): LatLng? {
        return milestone.location
    }

    override fun getTitle(): String? {
        return milestone.description
    }

    override fun getSnippet(): String? {
        return null
    }
}
